src/main/java/brandkon/product/ProductService.java

```diff
@@ -0,0 +1,62 @@
+package brandkon.product;
+
+import brandkon.brand.Brand;
+import brandkon.brand.BrandRepository;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class ProductService {
+
+    private ProductRepository productRepository;
+    private BrandRepository brandRepository;
+
+    public ProductService(ProductRepository productRepository, BrandRepository brandRepository) {
+        this.productRepository = productRepository;
+        this.brandRepository = brandRepository;
+    }
+
+    public List<ProductResponse> getAll(Long brandId) {
+
+        if (brandId == null) {
+            return mappingProductToResponse(productRepository.findAll());
+        }
+        return mappingProductToResponse(productRepository.findAllByBrandId(brandId));
+    }
+
+    private List<ProductResponse> mappingProductToResponse(List<Product> products) {
+        return products
+                .stream()
+                .map(product -> new ProductResponse(
+                        product.getId(),
+                        product.getBrand().getName(),
+                        product.getName(),
+                        product.getPrice(),
+                        product.getImageUrl())).toList();
+    }
+
+    public ProductDetailResponse get(Long productId) {
```

Comment:
이름을 좀 더 관습에 맞게 지어 주시면 더 좋을 것 같아요
```suggestion
    public ProductDetailResponse findById(Long productId) {
```